crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Repository
+    attr_reader :platform, :id, :registry, :available, :md5, :url
+
+    class << self
+      #
+      # class methods
+      #
+      def load_registry
+        @@config ||= YAML.load_file(File.join(Rails.root, "config", "repos.yml"))
+      end
+
+      def where(options = {})
+        platform = options.fetch :platform, nil
+        repo = options.fetch :repo, nil
+        check_all_repos.select do |r|
+          if platform
+            if repo
+              r.platform == platform && r.name == repo
+            else
+              r.platform == platform
+            end
+          else
+            r.name == repo
+          end
+        end
+      end
+
+      def all_platforms
+        @@config.keys
+      end
+
+      def check_all_repos
+        repochecks = []
+        all_platforms.each do |platform|
+          repositories(platform).each do |repo|
+            check = self.new(platform, repo)
+            repochecks << check
+          end
+        end
+        repochecks
+      end
+
+      def admin_platform
+        NodeObject.admin_node.target_platform
+      end
+
+      def web_port
+        Proposal.where(barclamp: "provisioner").first.properties["web_port"]
+      end
+
+      def repositories(platform)
+        @@config[platform]["repos"].keys
+      end
+    end
+
+    #
+    # constructor
+    #
+    def initialize(platform, repo)
+      @platform = platform
+      @id = repo
+      @registry = Repository.load_registry[@platform]["repos"][@id]
+      @available = valid_repo_dir?
+      @md5 = valid_key_file?
+      @url = repo_url
+    end
+
+    #
+    # instance methods
+    #
+    def valid_repo_dir?
+      all_repo_dirs.include?(@registry["name"])
+    end
+
+    def valid_key_file?
+      if File.exist?(repomd_key_path)
+        md5 = Digest::MD5.hexdigest(File.read(repomd_key_path))
+        expected_repomd_key_md5.include? md5
```

Comment:
No, actually I don't think it is necessary.  So ignore what I said before - instead can you just change `expected_repomd_key_md5` to be a `String` instead of an `Array`?